src/uikit.cs

```diff
@@ -26761,5 +27283,1203 @@ interface UITextInputContext {
+		nfloat CornerRadius { get; set; }
+
+		[Export ("maximumSize", ArgumentSemantic.Assign)]
+		CGSize MaximumSize { get; set; }
+
+		[Export ("accessibilityIgnoresInvertColors")]
+		bool AccessibilityIgnoresInvertColors { get; set; }
+	}
+
+	[NoWatch, TV (17, 0), iOS (17, 0)]
+	[BaseType (typeof (NSObject))]
+	interface UIContentUnavailableTextProperties : NSCopying, NSSecureCoding {
+		[Export ("font", ArgumentSemantic.Strong)]
+		UIFont Font { get; set; }
+
+		[Export ("color", ArgumentSemantic.Strong)]
+		UIColor Color { get; set; }
+
+		[Export ("lineBreakMode", ArgumentSemantic.Assign)]
+		UILineBreakMode LineBreakMode { get; set; }
+
+		[Export ("numberOfLines")]
+		nint NumberOfLines { get; set; }
+
+		[Export ("adjustsFontSizeToFitWidth")]
+		bool AdjustsFontSizeToFitWidth { get; set; }
+
+		[Export ("minimumScaleFactor")]
+		nfloat MinimumScaleFactor { get; set; }
+
+		[Export ("allowsDefaultTighteningForTruncation")]
+		bool AllowsDefaultTighteningForTruncation { get; set; }
+	}
+
+	[NoWatch, TV (17, 0), iOS (17, 0), MacCatalyst (17, 0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface UIContentUnavailableConfigurationState : UIConfigurationState {
+		[Export ("initWithTraitCollection:")]
+		[DesignatedInitializer]
+		NativeHandle Constructor (UITraitCollection traitCollection);
+
+		[NullAllowed, Export ("searchText", ArgumentSemantic.Strong)]
+		string SearchText { get; set; }
+	}
+
+	[NoWatch, TV (17, 0), iOS (17, 0), MacCatalyst (17, 0)]
+	[BaseType (typeof (UIView))]
+	[DisableDefaultCtor]
+	interface UIContentUnavailableView : UIContentView {
+		[Export ("initWithConfiguration:")]
+		[DesignatedInitializer]
+		NativeHandle Constructor (UIContentUnavailableConfiguration configuration);
+
+		[Export ("scrollEnabled")]
+		bool ScrollEnabled { [Bind ("isScrollEnabled")] get; set; }
+	}
+
+	[NoWatch, NoTV, iOS (17, 0), MacCatalyst (17, 0)]
+#if XAMCORE_5_0
+	[BaseType (typeof(UIViewController))]
+	interface UIDocumentViewController
+#else
+	[BaseType (typeof (UIViewController), Name = "UIDocumentViewController")]
+	interface UIDocViewController
+#endif
+	{
+
+		[DesignatedInitializer]
+		[Export ("initWithNibName:bundle:")]
+		[PostGet ("NibBundle")]
+		NativeHandle Constructor ([NullAllowed] string nibName, [NullAllowed] NSBundle bundle);
+
+		[Export ("initWithDocument:")]
+		NativeHandle Constructor ([NullAllowed] UIDocument document);
+
+		[NullAllowed, Export ("document", ArgumentSemantic.Strong)]
+		UIDocument Document { get; set; }
+
+		[Export ("navigationItemDidUpdate")]
+		void NavigationItemDidUpdate ();
+
+		[Async]
+		[Export ("openDocumentWithCompletionHandler:")]
+		void OpenDocument (Action<bool> completionHandler);
+
+		[Export ("documentDidOpen")]
+		void DocumentDidOpen ();
+
+		[Export ("undoRedoItemGroup")]
+		UIBarButtonItemGroup UndoRedoItemGroup { get; }
+	}
+
+	interface IUIShapeProvider { }
+
+	[NoWatch, NoTV, iOS (17, 0), MacCatalyst (17, 0)]
+	[Protocol]
+	interface UIShapeProvider {
```

Comment:
iirc protocols need both model and base type attributes?
```suggestion
	[NoWatch, NoTV, iOS (17, 0), MacCatalyst (17, 0)]
#if NET
	[Protocol, Model]
#else
	[Protocol, Model (AutoGeneratedName = true)]
#endif
	[BaseType (typeof (NSObject))]
	interface UIShapeProvider {
```